### PR TITLE
Reader: delete old related posts component

### PR DESCRIPTION
**Pull request by samouri**
Deleting some unused components we no longer need
1. Old related posts (not sure where this was used.  I think in the prefresh)
2. Reader author and site (was used in X-posts by replaced by ReaderAvatar)

### CODE REVIEW

**Comment by bluefuton** [comment]
Can we split this one into two? Sounds like two separate bits of cleanup, I reckon.

Perhaps keeping this PR for "remove old related posts" and  #12738 for the site-and-author-icon?

File: client/devdocs/design/blocks.jsx [hunk 1]
```diff
@@ -44,7 +44,6 @@ import ReaderAuthorLink from 'blocks/reader-author-link/docs/example';
 import ReaderSiteStreamLink from 'blocks/reader-site-stream-link/docs/example';
 import ReaderFullPostHeader from 'blocks/reader-full-post/docs/header-example';
 import AuthorCompactProfile from 'blocks/author-compact-profile/docs/example';
-import RelatedPostCard from 'blocks/reader-related-card/docs/example';
```
Comment (bluefuton) [inline client/devdocs/design/blocks.jsx]: Do we need to remove the `<RelatedPostCard />` lower down too?

Comment (samouri) [inline client/devdocs/design/blocks.jsx]: deleted!

**Comment by bluefuton** [comment]
> Can we split this one into two? Sounds like two separate bits of cleanup, I reckon.

Actually, these two dovetail quite nicely, as only the old related posts component uses `reader-author-and-site`. I take it back.

Me and @samouri just discovered that this is not related at all to #12738 for the site-and-author-icon removal. Just similarly-named components :)

**Pull request merged**